enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1293 @@
+The service object will look as follows:
+```sh
+ % oc get svc router-test -n openshift-ingress -o yaml
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-healthcheck-healthy-threshold: "2"
+    service.beta.kubernetes.io/aws-load-balancer-healthcheck-interval: "10"
+    service.beta.kubernetes.io/aws-load-balancer-healthcheck-timeout: "4"
+    service.beta.kubernetes.io/aws-load-balancer-healthcheck-unhealthy-threshold: "2"
+    service.beta.kubernetes.io/aws-load-balancer-type: nlb
+    traffic-policy.network.alpha.openshift.io/local-with-fallback: ""
+  creationTimestamp: "2024-04-10T02:07:29Z"
+  finalizers:
+  - service.kubernetes.io/load-balancer-cleanup
+  labels:
+    app: router
+    ingresscontroller.operator.openshift.io/owning-ingresscontroller: test
+    router: router-test
+  name: router-test
+  namespace: openshift-ingress
+  ownerReferences:
+  - apiVersion: apps/v1
+    controller: true
+    kind: Deployment
+    name: router-test
+    uid: 466ac236-050b-4583-b2d1-99323f31fcba
+  resourceVersion: "90321"
+  uid: 95d070ad-bbf0-45c9-b356-ed9db8d3cdae
+spec:
+  allocateLoadBalancerNodePorts: true
+  clusterIP: 172.30.230.109
+  clusterIPs:
+  - 172.30.230.109
+  externalTrafficPolicy: Local
+  healthCheckNodePort: 32652
+  internalTrafficPolicy: Cluster
+  ipFamilies:
+  - IPv4
+  ipFamilyPolicy: SingleStack
+  ports:
+  - name: http
+    nodePort: 30453
+    port: 80
+    protocol: TCP
+    targetPort: http
+  - name: https
+    nodePort: 32704
+    port: 443
+    protocol: TCP
+    targetPort: https
+  selector:
+    ingresscontroller.operator.openshift.io/deployment-ingresscontroller: test
+  sessionAffinity: None
+  type: LoadBalancer
+status:
+  loadBalancer:
+    ingress:
+    - hostname: a95d070adbbf045c9b356ed9db8d3cda-89fbf4dfc638408e.elb.us-east-1.amazonaws.com
+```
+
+
+### API Extensions
+- The API for setting `eipAllocations` field in the installer-config will look as follows in the file `pkg/types/aws/platform.go` of `openshift/installer` github repo:
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+...
+    // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/load-balancer-types.html#nlb 
+    // +optional 
+    LBType configv1.AWSLBType `json:"lbType,omitempty"`
+
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *configv1.AWSNetworkLoadBalancerParameters `json:"nlbParameters,omitempty"`
+}
+
+```
+
+- The installer will propagate the eipAllocations to the cluster ingress configuration which the ingress operator uses to create the default Ingress Controller
+  will look as follows in the file `config/v1/types_ingress.go` of `openshift/api` github repo:
```

Comment:
This is a run-on sentence (although Google Doc's grammar checker misses it).  Also, please make sure to use consistent formatting, and link to the relevant source code:
```suggestion
- The installer will propagate `eipAllocations` from `install-config.yaml` to the cluster ingress configuration, which the ingress operator uses to create the default IngressController.  
The following  definitions will be added in the file [`config/v1/types_ingress.go`](https://github.com/openshift/api/blob/master/config/v1/types_ingress.go) of the [`openshift/api`](https://github.com/openshift/api) GitHub repository: